### DISCUSSION TITLE
digital output not working constantly

### DISCUSSION THREAD

**Discussion by Kasho-2001** [bitcraze/discussions#145] [Q&A]
Hello everyone,
we work on the bolt drone and added a digital output (PB5) by using "params". So we have a python script and when we run it, the voltage of the pin should jump to 3 V.
In the beginning it worked quite well, but now it just does not work the most of the time. It is very strange, because there seems to be no rule whether it works or not.  We already checked and do not have any bad soldering points, it is also not a problem of falling voltage of the battery(we first thought so, because we just use 1s - 3.7V), It seems to work more often, if we wait about 2 minutes after switching on the bolt before running the script. All in all it is a very strange behavior and it is impossible to work with the digitial output. Hopefully there is a way to run it more reliable.

**Comment by knmcguire** [comment]
Hi! 

Unfortunately we have very little people at the office right now, and the one that is able to answer this will be gone for 4 months (@tobbeanton) . I'll keep this open for him but I hope that this issue is not blocking you right now?

**Reply by Kasho-2001** [reply]
Hi, okay we will find a way to overcome it. If we found out, i will post it here on github. Thanks for the quick reply.

**Reply by knmcguire** [reply]
Ah I wrote it wrongly here. Tobias is gone for only **4 weeks** not 4 months (not sure why I made that mistake). But still I hope you will be able to figure something out to cross that time. Sorry about that!

**Reply by Kasho-2001** [reply]
Hi!
small update, now i found the problem:

In "stabilizer.c", there is a line:   "while(!sensorsAreCalibrated()) {"
It showed, that the whole code is standing still, because it is stuck in this while loop, where it waits until the sensors are calibrated.
It is not an issue with a single bolt circuit board, we have this problem with all ones. 
So why the bolts need to calibrate so long and why they take always different amount of time?
Now we commented this while loop and the digital output works great, but due to not calibrated sensors, we can't fly them now :(
So maybe you have already made some experience with the calibration of the sensors and can help in this point.

**Reply by krichardsson** [reply]
Great that you found the root cause!
For the sensors to be calibrated the platform must not move for about a second, put it on a table for instance. This is the intended behavior, but it is also possible that there is a bug(!). I don't have a Bolt to test on but right now, but if you think you have found a bug, please file a github issue in the bitcraze/crazyflie-firmware repository and link to this discussion. 
Thanks!

**Reply by Kasho-2001** [reply]
HI, thanks for your response,
even if we had the drone completely stabilized calm and not tilted, we had sometimes very long calibration times. It seems to be completly random.
I am going to open a new github issue, thank you for the support:)

**Reply by tjt22** [reply]
Hi there! Did you ever post the code that activates the digital pin over python? I'm looking to do something similar, but don't know where to start.

**Reply by krichardsson** [reply]
Last comment duplicate of #893
